common/src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -509,6 +510,33 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
       .asBoolean()
   }
 
+  def validateQuery(query: String, expectedQueryTypes: org.neo4j.driver.summary.QueryType*): String = try {
+    val queryType = session.run(s"EXPLAIN $query").consume().queryType()
+    if (expectedQueryTypes.isEmpty || expectedQueryTypes.contains(queryType)) {
+      ""
+    } else {
+      s"Query non valid because the expected type should be [${expectedQueryTypes.mkString(", ")}], but the actual type is $queryType"
```

Comment:
what about including the query as well in the error message?
```suggestion
      s"Invalid query $query because the accepted types are [${expectedQueryTypes.mkString(", ")}], but the actual type is $queryType"
```

Reply:
Oki